src/Parser/Mage/Arcane/Modules/Features/ArcaneOrb.js

```diff
@@ -0,0 +1,109 @@
+
+class ArcaneOrb extends Analyzer {
+	static dependencies = {
+		abilityTracker: AbilityTracker,
+	};
+	
+	totalHits = 0;
+	badCasts = 0;
+	orbCast = true;
+
+	constructor(...args) {
+    super(...args);
+	   this.active = this.selectedCombatant.hasTalent(SPELLS.ARCANE_ORB_TALENT.id) && !this.selectedCombatant.hasShoulder(ITEMS.MANTLE_OF_THE_FIRST_KIRIN_TOR.id);
+  }
+
+	on_byPlayer_damage(event) {
+		const spellId = event.ability.guid;
+		if (spellId !== SPELLS.ARCANE_ORB_DAMAGE.id) {
+			return;
+		}
+		this.totalHits += 1;
+		this.orbCast = false;
+	}
+
+	on_byPlayer_cast(event) {
+		const spellId = event.ability.guid;
+		if (spellId !== SPELLS.ARCANE_ORB_TALENT.id) {
+			return;
+		}
+		if (this.orbCast === true) {
+			this.badCasts += 1;
+		}
+		this.orbCast = true;
+	}
+
+	on_finished() {
+		if (this.orbCast === true) {
+			this.badCasts += 1;
+		}
+	}
+
+	get averageHitPerCast() {
+		return this.totalHits / this.abilityTracker.getAbility(SPELLS.ARCANE_ORB_TALENT.id).casts;
+	}
+
+	get averageHitThresholds() {
+    return {
+      actual: this.averageHitPerCast,
+      isLessThan: {
+        minor: 2,
+        average: 1.5,
+        major: 1,
+      },
+      style: 'number',
+    };
+	}
+	
+	get badCastsThresholds() {
+    return {
+      actual: this.averageHitPerCast,
+      isGreaterThan: {
+        minor: 0,
+        average: 1,
+        major: 2,
+      },
+      style: 'number',
+    };
+  }
+
+	suggestions(when) {
+		when(this.averageHitThresholds)
+			.addSuggestion((suggest, actual, recommended) => {
+				return suggest(<React.Fragment>On average, your <SpellLink id={SPELLS.ARCANE_ORB_TALENT.id} /> hit ${formatNumber(this.averageHitPerCast,2)} times per cast. While it is beneficial to cast this even if it will only hit one mob, the talent is suited more towards AOE than Single Target. So if the fight is primarily Single Target, consider taking a different talent.</React.Fragment>)
+					.icon(SPELLS.ARCANE_ORB_TALENT.icon)
+					.actual(`${formatNumber(this.averageHitPerCast)} Hits Per Cast`)
+					.recommended(`${formatNumber(recommended)} is recommended`);
+			});
+		when(this.averageHitThresholds)
+		.addSuggestion((suggest, actual, recommended) => {
+			return suggest(<React.Fragment>On average, your <SpellLink id={SPELLS.ARCANE_ORB_TALENT.id} /> hit ${formatNumber(this.averageHitPerCast,2)} times per cast. While it is beneficial to cast this even if it will only hit one mob, the talent is suited more towards AOE than Single Target. So if the fight is primarily Single Target, consider taking a different talent.</React.Fragment>)
+				.icon(SPELLS.ARCANE_ORB_TALENT.icon)
+				.actual(`${formatNumber(this.averageHitPerCast)} Hits Per Cast`)
+				.recommended(`${formatNumber(recommended)} is recommended`);
+		});
+	}
+
+	statistic() {
+    return (
+			<StatisticBox
+  icon={<SpellIcon id={SPELLS.ARCANE_ORB_TALENT.id} />}
+  value={`${formatNumber(this.averageHitPerCast,2)}`}
+  label="Arcane Orb Hits Per Cast"
```

Comment:
Only the first word and names shound start with a Capital letter: `Arcane Orb hits per cast`